### PR TITLE
use SerializableIsolation for REPEATABLE READ

### LINKED ISSUE #3527: Snapshot Isolation does not support REPEATABLE READ
In [sql/parser/sql.y](https://github.com/cockroachdb/cockroach/blob/master/sql/parser/sql.y), there is the following text:

```
| REPEATABLE READ
  {
    // Mapped to the closest supported isolation level.
    $$ = SnapshotIsolation
  }
```

The text means that Snapshot Isolation supports REPEATABLE READ. But it is not true. In [1], there is the following text:

> Remark 9. REPEATABLE READ »« Snapshot Isolation.

The remark means that Snapshot Isolation and  REPEATABLE READ are incomparable. Since CockroachDB only supports Snapshot Isolation and SERIALIZABLE, SERIALIZABLE is needed to support REPEATABLE READ.

[1]: http://research.microsoft.com/pubs/69541/tr-95-51.pdf "A Critique of ANSI SQL Isolation Levels"

### PR DESCRIPTION
**Pull request by yaojingguo**
Fixes #3527

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3537)

### CODE REVIEW

**Comment by yaojingguo** [comment]
`sql.go` is generated by running `make` in `sql/parser` directory after `sql.y` is updated.

**Comment by tschottdorf** [comment]
LGTM. Thanks @yaojingguo!

File: sql/parser/sql.y [hunk 1]
```diff
@@ -861,14 +861,14 @@ iso_level:
...
+| SNAPSHOT
   {
-    // Mapped to the closest supported isolation level.
     $$ = SnapshotIsolation
   }
-| SNAPSHOT
+| REPEATABLE READ
   {
-    $$ = SnapshotIsolation
+    // Mapped to the closest supported isolation level.
```
Comment (petermattis) [inline sql/parser/sql.y]: Looking at this months after I wrote it, I wonder if it would be better to error when the user asks for `REPEATABLE READ`?

Comment (bdarnell) [inline sql/parser/sql.y]: Why make it an error? Our `SERIALIZABLE` provides a superset of the guarantees that the standard specifies for `REPEATABLE READ`, so we might as well support all four standard isolation levels with their closest equivalent. Other databases do the same, e.g. `READ UNCOMMITTED` is often implemented as a synonym for `READ COMMITTED` and no one minds that they never actually see dirty data.

Comment (petermattis) [inline sql/parser/sql.y]: Depends on what the user expectation is. If they are asking for `REPEATABLE READ` in order to get better performance wouldn't it be better to say, err, no we can't do that. If they are asking for `REPEATABLE READ` in order to get certain guarantees then swapping in `SERIALIZABLE` makes more sense.

Comment (bdarnell) [inline sql/parser/sql.y]: The SQL standard defines these isolation levels in terms of the minimum guarantees they provide; a conforming database could define all four as equivalent to `SERIALIZABLE`, but I think the standard requires that all four names be understood. The actual implementation of isolation levels vary so much from one DB to another that I don't think users can have any particular expectation of performance differences.

Comment (tschottdorf) [inline sql/parser/sql.y]: I also think that when you're fine tuning isolation levels for performance, then you've looked into what database you're tuning against. Agree with Ben that returning an error wouldn't change much except for making life harder for those users that run through a codebase which went wild with the use of exot-ish levels.

Comment (petermattis) [inline sql/parser/sql.y]: I don't feel strongly about this. Let's leave as-is.

Comment (JackKrupansky) [inline sql/parser/sql.y]: Would be good for any logging or UI display of queries to indicate the precise isolation semantics being used.

Comment (yaojingguo) [inline sql/parser/sql.y]: We have already used SNAPSHOT for READ UNCOMMITTED in a similar way. So leaving as-is is fine.

Comment (yaojingguo) [inline sql/parser/sql.y]: @JackKrupansky, there is already a way to show the precise isolation being actually used. `SHOW TRANSACTION ISOLATION LEVEL` does it:

```
SET TRANSACTION ISOLATION LEVEL READ COMMITTED

SHOW TRANSACTION ISOLATION LEVEL
----
SNAPSHOT
```

**Pull request merged**